packages/react/App.tsx

```diff
@@ -32,18 +32,20 @@ interface AppState {
 }
 
 export default class App extends React.Component<{}, AppState> {
-  buttonRef: React.RefObject<CdsButton>;
+  buttonRef: React.RefObject<typeof CdsButton & HTMLButtonElement>;
 
   constructor(props: any) {
     super(props);
     this.state = { modalOpen: false, modal2Open: false };
-    this.buttonRef = React.createRef<CdsButton>();
+    this.buttonRef = React.createRef<typeof CdsButton & HTMLButtonElement>();
   }
 
   componentDidMount() {
-    this.buttonRef.current.nativeElement.then(element => {
-      element.focus();
-    });
+    // (this.buttonRef.current as any).nativeElement.then((element: any) => {
+    //   element.focus();
+    // });
+
+    this.buttonRef.current.focus();
```

Comment:
Refs now work as expected. To prevent a breaking change there is still a `nativeElement` property available on the API.